Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -225,115 +230,67 @@ public class HTTPClient {
     ///
     /// - parameters:
     ///     - url: Remote URL.
+    ///     - context: Baggage context associated with this request
     ///     - deadline: Point in time by which the request must complete.
-    public func get(url: String, deadline: NIODeadline? = nil) -> EventLoopFuture<Response> {
-        return self.get(url: url, deadline: deadline, logger: HTTPClient.loggingDisabled)
-    }
-
-    /// Execute `GET` request using specified URL.
-    ///
-    /// - parameters:
-    ///     - url: Remote URL.
-    ///     - deadline: Point in time by which the request must complete.
-    ///     - logger: The logger to use for this request.
-    public func get(url: String, deadline: NIODeadline? = nil, logger: Logger) -> EventLoopFuture<Response> {
-        return self.execute(.GET, url: url, deadline: deadline, logger: logger)
+    public func get(url: String, context: BaggageContext, deadline: NIODeadline? = nil) -> EventLoopFuture<Response> {
+        return self.execute(.GET, url: url, context: context, deadline: deadline)
```

Comment:
for backward compatibility, the context should be optional or have default (static? "none") value?

Reply:
No, contexts should never be defaulted to nothing/empty - that leads to fragile APIs where dropping the traces becomes the default operation, rather than doing the right thing -- keep passing along contexts. We want our APIs to help people do the right thing. Note in a same way, Go's context values must _never_ be defaulted to empty, we're in the same boat.

It's another discussion _how and when_ this PR will be merged though; I do strongly believe contexts _must not_ be optional, as that leads to breaking traces accidentally (been there, done that). See discussions in https://forums.swift.org/t/server-distributed-tracing/37464/10 and the guidelines in https://github.com/slashmo/gsoc-swift-baggage-context#argument-namingpositioning

Reply:
Makes sense. This will, however, affect integration in a big way.

Please see my comment about an integration branch below.

Reply:
Sure it will, it breaks API, we're aware.